File: salao-frontend/src/services/notaFiscalSaidaService.ts
import axios from 'axios';

const API = 'http://localhost:8080/api';

export interface NotaFiscalSaida {
  id: number;
  numeroNota: string;
  serie?: string;
  dataEmissao: string;
  cliente?: { id: number; nome: string };
  valorTotal: number;
  transportadoraNome?: string;
  veiculoPlaca?: string;
  vendaId?: number;
}

export interface TransporteRequest {
  transportadoraNome?: string;
  veiculoPlaca?: string;
}

export const notaFiscalSaidaService = {
  listar:              ()                                     => axios.get<NotaFiscalSaida[]>(`${API}/notas-fiscais-saida`),
  buscarPorId:         (id: number)                          => axios.get<NotaFiscalSaida>(`${API}/notas-fiscais-saida/${id}`),
  atualizarTransporte: (id: number, data: TransporteRequest) => axios.put(`${API}/notas-fiscais-saida/${id}/transporte`, data),
};

File: salao-backend/src/main/java/com/salao/modules/fiscal/saida/NotaFiscalSaidaService.java
package com.salao.modules.fiscal.saida;

import com.salao.modules.fiscal.entrada.TransporteRequestDTO;
import com.salao.modules.venda.Venda;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class NotaFiscalSaidaService {

    private final NotaFiscalSaidaRepository repository;

    public List<NotaFiscalSaidaResponseDTO> listar() {
        return repository.findAll().stream().map(NotaFiscalSaidaResponseDTO::from).toList();
    }

    public NotaFiscalSaidaResponseDTO buscarPorId(Long id) {
        return NotaFiscalSaidaResponseDTO.from(buscarEntidade(id));
    }

    @Transactional
    public NotaFiscalSaida criarParaVenda(Venda venda) {
        var nota = repository.save(NotaFiscalSaida.builder()
                .venda(venda)
                .cliente(venda.getCliente())
                .valorTotal(venda.getValorTotal())
                .serie("1")
                .build());
        nota.setNumeroNota("NFS-" + String.format("%06d", nota.getId()));
        return repository.save(nota);
    }

    public NotaFiscalSaidaResponseDTO atualizarTransporte(Long id, TransporteRequestDTO dto) {
        var nota = buscarEntidade(id);
        nota.setTransportadoraNome(dto.transportadoraNome());
        nota.setVeiculoPlaca(dto.veiculoPlaca());
        return NotaFiscalSaidaResponseDTO.from(repository.save(nota));
    }

    private NotaFiscalSaida buscarEntidade(Long id) {
        return repository.findById(id)
                .orElseThrow(() -> new RuntimeException("Nota fiscal de saída não encontrada"));
    }
}
